src/CommitmentsV2/SFA.DAS.CommitmentsV2/Application/Commands/BulkUploadValidateRequest/ValidateCourseCode.cs

```diff
@@ -27,8 +27,26 @@ private List<Error> ValidateCourseCode(BulkUploadAddDraftApprenticeshipRequest c
             {
                 domainErrors.Add(new Error("CourseCode", "Enter a valid <b>standard code</b>"));
             }
+            else if (providerStandardResults.IsMainProvider && IsValidMainProviderStandardDetails(csvRecord.CourseCode, providerStandardResults) != true)
+            {
+                domainErrors.Add(new Error("CourseCode", "Enter a valid <b>standard code.</b> You have not told us that you deliver this training course. You must assign the course to your account in the <a href=" + _urlHelper.CourseManagementLink($"{csvRecord.ProviderId}/review-your-details") + " class='govuk - link'>Your standards and training venues</a> section."));
+
+            }
 
             return domainErrors;
         }
+
+        private bool IsValidMainProviderStandardDetails(string stdCode, ProviderStandardResults providerStandardResults)
+        {
+            if (string.IsNullOrWhiteSpace(stdCode)) return false;
+            if (providerStandardResults.Standards == null) return false;
```

Comment:
Won't standards be null only if the provider is not a Main provider, and we won't invoke this method in that case I don't think?

Reply:
Yes you are right, we wont invoke this method if the provider is not a main provider because the first condition will be false. Here it is handled just to show the validation message in case if the provider is a main provider and standards is null.